egs/wsj/s5/utils/prepare_lang_subword.py

```diff
@@ -0,0 +1,487 @@
+
+cat $tmpdir/align_lexicon.txt | \
+  perl -ane '@A = split; print $A[0], " ", join(" ", @A), "\n";' | sort | uniq > $dir/phones/align_lexicon.txt
+
+if [ -f $srcdir/nonterminals.txt ]; then
+#  utils/lang/grammar/augment_phones_txt.py $dir/phones.txt $srcdir/nonterminals.txt $dir/phones.txt
+#  utils/lang/grammar/augment_words_txt.py $dir/words.txt $srcdir/nonterminals.txt $dir/words.txt
+#  cp $srcdir/nonterminals.txt $dir/phones/nonterminals.txt
+#  utils/sym2int.pl $dir/phones.txt <$dir/phones/nonterminals.txt >$dir/phones/nonterminals.int
+#
+#  for w in "#nonterm_begin" "#nonterm_end" $(cat $srcdir/nonterminals.txt); do
+#    echo $w $w  # These are words without pronunciations, so leave those prons
+#                # empty.
+#  done >> $dir/phones/align_lexicon.txt
+#  nonterm_phones_offset=$(grep '#nonterm_bos' <$dir/phones.txt | awk '{print $2}')
+#  echo $nonterm_phones_offset > $dir/phones/nonterm_phones_offset.int
+#  echo '#nonterm_bos' > $dir/phones/nonterm_phones_offset.txt  # temporary.
+#
+#  if [ -f $dir/phones/word_boundary.txt ]; then
+#    # word-position-dependent system.  Only include the optional-silence phone,
+#    # and phones that can end a word, plus the special symbol #nonterm_bos, in the
+#    # left-context phones.
+#    awk '{if ($2 == "end" || $2 == "singleton") print $1; }' <$dir/phones/word_boundary.txt | \
+#        cat - $dir/phones/optional_silence.txt $dir/phones/nonterm_phones_offset.txt > $dir/phones/left_context_phones.txt
+#  else
+#    cat $dir/phones/{silence,nonsilence}.txt $dir/phones/nonterm_phones_offset.txt > $dir/phones/left_context_phones.txt
+#  fi
+#  utils/sym2int.pl $dir/phones.txt <$dir/phones/left_context_phones.txt >$dir/phones/left_context_phones.int
+#
+#  # we need to write utils/lang/make_lexicon_fst_silprob.py before this can work.
+#  grammar_opts="--left-context-phones=$dir/phones/left_context_phones.txt --nonterminals=$srcdir/nonterminals.txt"
+#else
+  echo "Currently do not support nonterminals"
+  grammar_opts=
+fi
+
+# create phones/align_lexicon.int from phones/align_lexicon.txt
+cat $dir/phones/align_lexicon.txt | utils/sym2int.pl -f 3- $dir/phones.txt | \
+  utils/sym2int.pl -f 1-2 $dir/words.txt > $dir/phones/align_lexicon.int
+
+# Create the basic L.fst without disambiguation symbols, for use
+# in training.
+
+if $silprob; then
+#  # Add silence probabilities (models the prob. of silence before and after each
+#  # word).  On some setups this helps a bit.  See utils/dict_dir_add_pronprobs.sh
+#  # and where it's called in the example scripts (run.sh).
+#  utils/lang/make_lexicon_fst_silprob.py $grammar_opts --sil-phone=$silphone \
+#         $tmpdir/lexiconp_silprob.txt $srcdir/silprob.txt | \
+#     fstcompile --isymbols=$dir/phones.txt --osymbols=$dir/words.txt \
+#       --keep_isymbols=false --keep_osymbols=false |   \
+#     fstarcsort --sort_type=olabel > $dir/L.fst || exit 1;
+  echo "Currently do not support word-dependnet silence probability"
```

Comment:
we do not support word-dependent.
and you should start the echo statement echo "$0:", and do
exit 1
after it.